src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/nowhitespacebeforecasedefaultcolon/Example1.java

```diff
@@ -0,0 +1,38 @@
+/*
+NoWhitespaceBeforeCaseDefaultColon
+
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.whitespace.nowhitespacebeforecasedefaultcolon;
+
+class Example1 {
+  void example() {
+    switch(1) {
+      case 1 : // violation '':' is preceded with whitespace'
+          break;
+      case 2: 
+          break;
+      default : // violation '':' is preceded with whitespace'
+          break;
+    }
+
+    switch(2) {
+      case 2:
+          break;
+      case 3, 4
+              : break; // violation '':' is preceded with whitespace'
```

Comment:
I have issues with this line
> Enhanced 'switch' blocks are not supported at language level '11'

Does that mean I have to move the example to `resources-noncompilable` and then invoke file via `getNonCompilablePath()`? 

I think the method has to be refactored though, because it is hardcoded to `src/test`. Am I right? Somebody please confirm.

https://github.com/checkstyle/checkstyle/blob/05444969f24969f2c9e28bd6cb53ad6a06dac342/src/test/java/com/puppycrawl/tools/checkstyle/AbstractModuleTestSupport.java#L174-L176

Reply:
Examples must be compileable, please change example to confirm to this simple rule,.

It is ok that example content can not be same as it was before.

Reply:
If any craziness happens in Example code, please feel free to change it up to your vision.
Just note it in PR description that you did extra changes due to .....

Reply:
Done. Changed.

Reply:
We are only delaying the inevitable; we have some checks that require higher language levels (RecordComponentName comes to mind). Let's not bind ourselves to using ancient code elsewhere. If anything, we should be trying to include newer syntax when possible, not avoiding it.

Reply:
@nrmancuso Copied example exactly and moved to `resources-noncompilable`. Should be good.

Reply:
Example must be compileable at least by some jdk.
Is examples folder grepped for jdk compilation in CI ?

Reply:
https://github.com/checkstyle/checkstyle/pull/13243

Reply:
@romani assigned back to you to make sure you are good with doing check for compilation comment and compilation in CI later.

Reply:
Yes, I trust @stoyanK7 , that we will not drop issue without attention, and will be fixed some time later.